snuba/migrations/runner.py

```diff
@@ -434,3 +434,69 @@ def add_node(
             elif isinstance(migration, CodeMigration):
                 for python_op in migration.forwards_global():
                     python_op.execute_new_node(storage_sets)
+
+    @classmethod
+    def copy_tables(
+        self,
+        node_type: ClickhouseNodeType,
+        host_name: str,
+        port: int,
+        user: str,
+        password: str,
+        database: str,
+        new_port: int,
+        table_names: Optional[str],
+        dry_run: bool = False,
+    ) -> None:
+        """
+        Create tables on a new node in a cluster by using the existing create table
+        statement from an existing node in a cluster. This can be done for a new shard
+        or new replica.
+        """
+        client_settings = ClickhouseClientSettings.QUERY.value
+        clickhouse = ClickhousePool(
+            host_name,
+            port,
+            user,
+            password,
+            database,
+            client_settings=client_settings.settings,
+            send_receive_timeout=client_settings.timeout,
+        )
+
+        # TODO: assert that the new node is not in the system.clusters definitions yet
```

Comment:
Did we discuss this before? Not sure why we want to enforce this. 
Did you discuss this process with @nikhars  and @mwarkentin ? 
We want to make the addition of a new node quick, that means having as few manual steps as possible. adding the tables before adding the node to the cluster implies at least three steps: provision node, add tables, then add the node to the cluster. 

I think we should ensure we are on the same page on the quickest process to add a node.